File: src/trithemius/trithemius.c
#include <stdio.h>
#include <ctype.h>

void trithemius(char *text, int encrypt) {
    for (int i = 0; text[i] != '\0'; i++) {
        if (isalpha(text[i])) {
            char start = isupper(text[i]) ? 'A' : 'a';
            int shift = encrypt ? i : -i;
            
            // Handle negative results for decryption by adding 26 before modulo
            text[i] = (char)((((text[i] - start) + shift) % 26 + 26) % 26 + start);
        }
    }
}

int main() {
    char message[] = "Hello World";
    
    printf("Original: %s\n", message);
    
    trithemius(message, 1);
    printf("Encrypted: %s\n", message);
    
    trithemius(message, 0);
    printf("Decrypted: %s\n", message);
    
    return 0;

File: src/trithemius/trithemius.py
def trithemius_cipher(text, encrypt=True):
    result = ""
    for i, char in enumerate(text):
        if char.isalpha():
            # Determine starting point (A=65, a=97)
            start = ord('A') if char.isupper() else ord('a')
            # Shift increases by the index 'i'
            shift = i if encrypt else -i
            # Calculate new character
            new_char = chr((ord(char) - start + shift) % 26 + start)
            result += new_char
        else:
            result += char
    return result

# Practice Test
message = "Hello World"
encrypted = trithemius_cipher(message)
print(f"Encrypted: {encrypted}")
